mmaction/models/localizers/ssn.py

```diff
@@ -0,0 +1,122 @@
+import torch
+import torch.nn as nn
+
+from .. import builder
+from ..registry import LOCALIZERS
+from .base import BaseLocalizer
+
+
+@LOCALIZERS.register_module
+class SSN(BaseLocalizer):
+
+    def __init__(self,
+                 backbone,
+                 cls_head,
+                 in_channels=3,
+                 spatial_type='avg',
+                 dropout_ratio=0.5,
+                 loss_cls=dict(type='SSNLoss'),
+                 train_cfg=None,
+                 test_cfg=None):
+
+        super(SSN, self).__init__()
+        self.backbone = builder.build_backbone(backbone)
+        self.cls_head = builder.build_head(cls_head)
+        self.is_test_prepared = False
+        self.in_channels = in_channels
+
+        self.spatial_type = spatial_type
+        if self.spatial_type == 'avg':
+            self.avg_pool = nn.AdaptiveAvgPool2d((1, 1))
+        elif self.spatial_type == 'max':
+            self.avg_pool = nn.AdaptiveMaxPool2d((1, 1))
+        else:
+            self.avg_pool = None
+
+        self.dropout_ratio = dropout_ratio
+        if self.dropout_ratio != 0:
+            self.dropout = nn.Dropout(p=self.dropout_ratio)
+        self.loss_cls = builder.build_loss(loss_cls)
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+        self.init_weights()
+
+    def init_weights(self):
+        super(SSN, self).init_weights()
+        self.backbone.init_weights()
+        self.cls_head.init_weights()
+
+    def extract_feat(self, imgs):
+        """Extract features through a backbone.
+
+        Args:
+            imgs (torch.Tensor): The input images.
+        Returns:
+            torch.tensor: The extracted features.
+        """
+        x = self.backbone(imgs)
+        return x
```

Comment:
`extract_feat` can be defined in `base.py`

Reply:
But BSN and BMN are based on feature directly, they do not need the backbone to extract features.

Reply:
So you can remove `super().__init__()` in `BMN` and `BSN`, since original `BaseLocalizer` does nothing. you can try it.